### PR TITLE
Add fr ca date time provider

### PR DESCRIPTION
**Pull request by davebulaval**
See issue #1764 for detail.

### CODE REVIEW

**Comment by davebulaval** [comment]
I was unsure if you wanted to duplicate the `TestFrFr` for coverage purposes. I think the less we can do is to add a mention of the shared format. Also, I have renamed the `TestFrFr` to `TestFrFrAndFrCa` to reflect this shared format.

**Comment by davebulaval** [comment]
Not quite sure of the failure reason.

**Review by fcurella — CHANGES_REQUESTED** [review]

File: tests/providers/test_date_time.py [hunk 1]
```diff
@@ -1195,7 +1195,8 @@ def test_month(self):
         assert month in NoNoProvider.MONTH_NAMES.values()
 
 
-class TestFrFr(unittest.TestCase):
+class TestFrFrAndFrCa(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (davebulaval) [reply tests/providers/test_date_time.py:1198]: Perfect. I will duplicate the tests later today.

[the remark this answers is not in the document]
Comment (davebulaval) [reply tests/providers/test_date_time.py]: @fcurella done!

**Review by fcurella — APPROVED** [review]

**Review by fcurella — APPROVED** [review]
Thank you!

**Pull request merged**